### PR TITLE
feat(bridge): find a relay point when a point-to-point link is too weak

### PR DESCRIPTION
**Pull request by gregbaase**
## Summary

When a **Point-to-point link** comes back **below sensitivity** (the transmitter and target can't talk directly), the planner now helps find where to put **one** extra node to relay between them.

The link card gains a **"Find bridge"** section (shown only when the link is too weak) that:
- runs coverage at **both** endpoints using the site's own radio + receiver values,
- highlights every cell where a single added node can hear **both** (the **coverage overlap**),
- marks the **single best** relay spot (the cell maximizing the weaker signal),
- reports the bridgeable area, best coordinates, signal strength and margin.

## Motivation

Field planning often ends with "the direct path is blocked — can I throw one relay node at it?" This answers that question with the exact same radio/terrain model the app already runs, reusing the point-to-point link's transmitter + target. No second simulation required.

## How it works

- The **point-to-point link** section is unchanged: pick the transmitter and a target.
- When the link verdict is below sensitivity, a relay section appears.
- **Find bridge** computes a coverage grid at the transmitter (A) and at the target (B) with the identical engine, then computes the overlap: every cell where `A >= rx_sensitivity` AND `B >= rx_sensitivity`.
- Because links are modeled symmetrically, a node in the overlap can hear both — that's the bridgeable region.
- Best placement = the overlap cell maximizing `min(A, B)`.

## Changes

- `src/bridge.ts` — new pure module: `computeBridge()` returns the overlap grid (shaped like a `CoverageResult` so it drops straight into the existing render pipeline), the best placement, area, margin, and a **direct-link** flag. DOM-free and unit-testable, mirroring `coverageStats.ts` / `coverageContours.ts`.
- `src/components/PointToPoint.vue` — relay section shown only when the direct link is below sensitivity; **Find bridge / Clear** + result states (direct link / bridge possible / no single relay).
- `src/store.ts` — `findBridge` / `clearBridge` + overlay & marker sync; the two coverage runs are **serial** (the engine is a single-instance worker pool); the relay-zone layer stays on top after visibility toggles.
- `src/map/overlay.ts` — `bridgeSelectionImage()` renders the relay zone as a **distinct green** layer (a different hue from the normal coverage heatmaps).
- `src/layers.ts` — `bridgePinElement()`: a yellow teardrop with a Venn **inner-join** glyph (A=yellow, overlap=black, B=yellow) for the recommended relay spot.
- `test/bridge.test.ts` — unit tests for `computeBridge` (full/empty overlap, min-scoring of best placement, non-aligned grid resampling, direct-link detection).

## Testing

- `pnpm test` — passes (new `bridge.test.ts` plus existing suites).
- `pnpm build` (`vue-tsc -b && vite build`) — passes.
- Manually verified: a weak link shows the relay section and draws the green overlap + best pin; a viable link does not show the section.

## Notes / assumptions

- The relay section only appears when the link is below sensitivity — the case where a bridge is actually useful.
- "A node in the overlap bridges the two" assumes the relay's receiver is modeled at each endpoint's threshold (consistent with how coverage and links are treated elsewhere).
- Overlap precision is bounded by the coarser of the two grids (mixed resolutions are resampled to the first).

## Summary by CodeRabbit

* **New Features**
  * Added bridge planning when direct signal is insufficient.
  * Identifies the strongest relay location, bridgeable area, and signal margin.
  * Added map overlays highlighting viable relay zones and suggested relay locations.
  * Added controls to calculate, cancel, and clear bridge analysis.
  * Reports direct-link availability or when no suitable relay can be found.

* **Bug Fixes**
  * Improved handling of missing coverage, invalid grids, and unavailable overlap areas.

### CODE REVIEW

File: src/store.ts [hunk 1]
```diff
@@ -639,6 +651,120 @@ const useStore = defineStore('store', {
...
+        this.bridgeError = error instanceof Error ? error.message : String(error);
+        this.bridgeState = 'error';
+      }
+    },
+    clearBridge() {
+      bridgeAbort?.abort();
+      this.bridgeResult = null;
+      this.bridgeState = 'idle';
+      this.bridgeError = '';
+      this.removeBridgeOverlay();
```
[the remark this answers is not in the document]
Comment (gregbaase) [reply src/store.ts:771]: resolved via the prompt and tested manually with success.

**Comment by mrpatrick1991** [comment]
> Best placement = the overlap cell maximizing min(A, B). 

It's an idea, but probably won't produce a realistic result (fresnel zones, etc). 

> Field planning often ends with "the direct path is blocked — can I throw one relay node at it?"

No, it doesn't. This is partially what area mode is for. 

In practice, 99.9% of the effort in placing nodes is based on selecting between locations where you have physical access and permission to do the installation. Knowing that there's a particular location where you could, theoretically, bridge two spots, isn't that useful, since with probability ~1.0 it won't be accessible. In any case, you could find such spots using the existing area mode functionality, and it would be even more useful, since you'd also see the relative coverage across space for your proposed relay. 

I don't think this will help the majority of users, and adds complexity to a tool which we want to keep simple.

File: src/bridge.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+ * longitude safely. Returns NaN when the point is outside B's grid.
+ * Only used when A and B are NOT aligned (the common case is aligned).
+ */
+function sampleNearest(result: CoverageResult, lat: number, lon: number): number {
+  const { bounds, width, height, pixelDegrees } = result;
+  if (lat > bounds.north || lat < bounds.south) return NaN;
+  const span = bounds.east - bounds.west;
+  let frac = (lon - bounds.west) / span;
+  frac = ((frac % 1) + 1) % 1; // wrap into [0,1) (handles antimeridian-crossing rasters)
+  const col = Math.min(width - 1, Math.round(frac * (width - 1)));
```
[the remark this answers is not in the document]
Comment (mrpatrick1991) [reply src/bridge.ts]: I'd want to see test cases for this. Right now it looks like there are edge cases where this can fail.

File: test/bridge.test.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      4, 4, 0.005
+    );
+    const r = computeBridge({ resultA: a, resultB: b, thresholdA: THRESHOLD, thresholdB: THRESHOLD, ...TX });
+    expect(r.hasOverlap).toBe(true);
+    expect(r.areaKm2).toBeCloseTo(4.96, 0); // all 4 coarse cells overlap
+    expect(Array.from(r.overlap.dbm)).toEqual([-80, -80, -80, -80]);
+  });
+
+  it('reports a direct link (no bridge needed) when each site hears the other', () => {
+    // Both transmitters sit in the centre where both grids are strong (-80 ≥ -130).
```
Comment (mrpatrick1991) [inline test/bridge.test.ts]: heuristics make no sense to me.

File: test/bridge.test.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    // Overlap grid: only top-left populated.
+    expect(r.overlap.dbm[0]).toBeCloseTo(-80, 6);
+    expect(Number.isNaN(r.overlap.dbm[1])).toBe(true);
+    expect(Number.isNaN(r.overlap.dbm[2])).toBe(true);
+    expect(Number.isNaN(r.overlap.dbm[3])).toBe(true);
+  });
+
+  it('best placement maximizes min(A,B)', () => {
+    // A strong everywhere. B strongest at top-left, weaker elsewhere.
+    const a = grid([-110, -110, -110, -110], 2, 2);
```
Comment (mrpatrick1991) [inline test/bridge.test.ts:70]: this is wrong. Why are we using a literal value for "strong" pinned to -110?

**Review by mrpatrick1991 — COMMENTED** [review]
suggest that we don't merge this in its current form.

**Comment by mrpatrick1991** [comment]
Closing this for now. 

Feel free to re-open when the above concerns are addressed.
